my-app/src/components/GenericForm.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import { Box } from "@mui/material";
+import FormHelperText from "@mui/material/FormHelperText";
+import Ajv, { JSONSchemaType, Schema } from "ajv";
+import ajvErrors from "ajv-errors";
+
+const ajv = new Ajv({ allErrors: true });
+require("ajv-errors")(ajv);
+
+interface Field {
+  id: string;
+  label: string;
+  type: string;
+  required: boolean;
+  placeholder: string;
+  errorMsg: string;
+  errorMsgVisibility: boolean;
+  options?: { value: string; label: string }[]; // For select fields
+}
+
+interface Props {
+  fields: Field[];
+  customSubmitFunction: (data: Record<string, any>) => void;
+  submitButtonName: string;
+  schema: Schema;
+}
+
+const GenericForm: React.FC<Props> = ({
+  fields,
+  customSubmitFunction,
+  submitButtonName,
+  schema,
+}) => {
+  const [formData, setFormData] = useState<Record<string, string | undefined>>(
+    {}
+  );
+  const [fieldsData, setFieldsData] = useState(fields);
+  const [currentLabelStyle, setCurrentLabelStyle] =
+    useState("signinLabelNormal");
+  const validate = ajv.compile(schema);
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    formState: { errors },
+  } = useForm();
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { id, value } = event.target;
+    setFormData({ ...formData, [id]: value });
+  };
+
+  const onChange = () => {
+    setCurrentLabelStyle("signinLabelNormal");
+  };
+  const validateForm = (data: Record<string, any>) => {
+    fields.map((field) => (field.errorMsgVisibility = true)); //Reset Error Messages
+    validate(data);
+    const errorMessages = Array.isArray(validate.errors) ? validate.errors : []; //Get all the errors
+    fields.map((field /*Go throu every field*/) =>
+      errorMessages.map(
+        (
+          message //Go throu every error
+        ) =>
+          message.instancePath.substring(1) == field.id //check if it is assign to the field id
+            ? (field.errorMsgVisibility = false) //if they match, show the error
+            : console.log()
+      )
+    );
+    setFieldsData(fields);
+  };
+
+  const onSubmit = async (data: Record<string, any>) => {
+    validateForm(data);
+    customSubmitFunction(data);
+  };
+
+  return (
+    <Box
+      component="form"
+      onSubmit={handleSubmit(onSubmit)}
+      sx={{ mt: 1 }}
+      onChange={onChange}
+    >
+      {fieldsData.map((field) => (
+        <Box key={field.id}>
+          <h4>{field.label}</h4>
+          {field.type !== "select" ? (
```

Comment:
Lets say we want to add a Switch field (boolean) "Yes" / "No" 
![image](https://github.com/HelloIAmBanana/bank-hafifa/assets/33515941/405c1f0d-79d2-4121-8597-e8bbacc2396b)
You will have to add another ternary operation which will :
1.  Make the code a ternary operators hell
2.  Hard to maintain and read .

So what I would ask you to do is to create a registry file which will have a object variable with 
{fieldName: SwitchField} (for example)

Than you will just use it this way fieldsRegistry[fieldName] and you will get the needed field .
It will be easy to maintain because to add a field you'l just need to add a key value to the object .